This notebook contains the investigation of the data that generated the preprocessing steps that are contained within the `process_data.py` script.

In [54]:
import pandas as pd
import numpy as np

import sqlalchemy
import os

# Raw input

## Messages

In [3]:
raw_messages = pd.read_csv('../data/disaster_messages.csv')

In [4]:
raw_messages.head()

id                                            message  \
0   2  Weather update - a cold front from Cuba that c...   
1   7            Is the Hurricane over or is it not over   
2   8                    Looking for someone but no name   
3   9  UN reports Leogane 80-90 destroyed. Only Hospi...   
4  12  says: west side of Haiti, rest of the country ...   

                                            original   genre  
0  Un front froid se retrouve sur Cuba ce matin. ...  direct  
1                 Cyclone nan fini osinon li pa fini  direct  
2  Patnm, di Maryani relem pou li banm nouvel li ...  direct  
3  UN reports Leogane 80-90 destroyed. Only Hospi...  direct  
4  facade ouest d Haiti et le reste du pays aujou...  direct

In [5]:
raw_messages.shape

(26248, 4)

## Categories

In [6]:
raw_categories = pd.read_csv('../data/disaster_categories.csv')

In [7]:
raw_categories.head()

id                                         categories
0   2  related-1;request-0;offer-0;aid_related-0;medi...
1   7  related-1;request-0;offer-0;aid_related-1;medi...
2   8  related-1;request-0;offer-0;aid_related-0;medi...
3   9  related-1;request-1;offer-0;aid_related-1;medi...
4  12  related-1;request-0;offer-0;aid_related-0;medi...

In [8]:
raw_categories.shape

(26248, 2)

Categories will be preprocessed before joining to set categories

### Fetching categories names

In [9]:
expanded_categories =  raw_categories['categories'].str.split(';', expand = True)

In [10]:
# Extracting categories

for i, value in enumerate(expanded_categories.iloc[0]):

    cat_name = value.split('-')[0]

    if i == 0:

        categories_name = [cat_name]

    else:

        categories_name.append(cat_name)

In [11]:
categories_name[0:5]

['related', 'request', 'offer', 'aid_related', 'medical_help']

In [12]:
expanded_categories.columns = categories_name

In [13]:
def get_category_label(frame, cat_cols):

    for col in frame[cat_cols].columns:

        frame[col] = frame[col].str.split('-').str[1]

        frame[col] = frame[col].astype(int)

    return frame

In [14]:
get_category_label(expanded_categories, expanded_categories.columns)

related  request  offer  aid_related  medical_help  medical_products  \
0            1        0      0            0             0                 0   
1            1        0      0            1             0                 0   
2            1        0      0            0             0                 0   
3            1        1      0            1             0                 1   
4            1        0      0            0             0                 0   
...        ...      ...    ...          ...           ...               ...   
26243        0        0      0            0             0                 0   
26244        0        0      0            0             0                 0   
26245        1        0      0            0             0                 0   
26246        1        0      0            1             0                 0   
26247        1        0      0            0             0                 0   

       search_and_rescue  security  military  child_alone  ...  aid_centers  \
0                      0         0         0            0  ...            0   
1                      0         0         0            0  ...            0   
2                      0         0         0            0  ...            0   
3                      0         0         0            0  ...            0   
4                      0         0         0            0  ...            0   
...                  ...       ...       ...          ...  ...          ...   
26243                  0         0         0            0  ...            0   
26244                  0         0         0            0  ...            0   
26245                  0         0         0            0  ...            0   
26246                  0         0         1            0  ...            0   
26247                  0         0         0            0  ...            0   

       other_infrastructure  weather_related  floods  storm  fire  earthquake  \
0                         0                0       0      0     0           0   
1                         0                1       0      1     0           0   
2                         0                0       0      0     0           0   
3                         0                0       0      0     0           0   
4                         0                0       0      0     0           0   
...                     ...              ...     ...    ...   ...         ...   
26243                     0                0       0      0     0           0   
26244                     0                0       0      0     0           0   
26245                     0                0       0      0     0           0   
26246                     0                0       0      0     0           0   
26247                     0                0       0      0     0           0   

       cold  other_weather  direct_report  
0         0              0              0  
1         0              0              0  
2         0              0              0  
3         0              0              0  
4         0              0              0  
...     ...            ...            ...  
26243     0              0              0  
26244     0              0              0  
26245     0              0              0  
26246     0              0              0  
26247     0              0              0  

[26248 rows x 36 columns]

In [15]:
raw_categories = pd.concat([raw_categories['id'], expanded_categories], axis = 1)

In [16]:
raw_categories.head()

id  related  request  offer  aid_related  medical_help  medical_products  \
0   2        1        0      0            0             0                 0   
1   7        1        0      0            1             0                 0   
2   8        1        0      0            0             0                 0   
3   9        1        1      0            1             0                 1   
4  12        1        0      0            0             0                 0   

   search_and_rescue  security  military  ...  aid_centers  \
0                  0         0         0  ...            0   
1                  0         0         0  ...            0   
2                  0         0         0  ...            0   
3                  0         0         0  ...            0   
4                  0         0         0  ...            0   

   other_infrastructure  weather_related  floods  storm  fire  earthquake  \
0                     0                0       0      0     0           0   
1                     0                1       0      1     0           0   
2                     0                0       0      0     0           0   
3                     0                0       0      0     0           0   
4                     0                0       0      0     0           0   

   cold  other_weather  direct_report  
0     0              0              0  
1     0              0              0  
2     0              0              0  
3     0              0              0  
4     0              0              0  

[5 rows x 37 columns]

# Generating Full raw dataframe

In [17]:
df_raw = raw_messages.merge(raw_categories,
                            on = 'id',
                            how = 'left',
                            indicator=True)

In [18]:
df_raw.head()

id                                            message  \
0   2  Weather update - a cold front from Cuba that c...   
1   7            Is the Hurricane over or is it not over   
2   8                    Looking for someone but no name   
3   9  UN reports Leogane 80-90 destroyed. Only Hospi...   
4  12  says: west side of Haiti, rest of the country ...   

                                            original   genre  related  \
0  Un front froid se retrouve sur Cuba ce matin. ...  direct        1   
1                 Cyclone nan fini osinon li pa fini  direct        1   
2  Patnm, di Maryani relem pou li banm nouvel li ...  direct        1   
3  UN reports Leogane 80-90 destroyed. Only Hospi...  direct        1   
4  facade ouest d Haiti et le reste du pays aujou...  direct        1   

   request  offer  aid_related  medical_help  medical_products  ...  \
0        0      0            0             0                 0  ...   
1        0      0            1             0                 0  ...   
2        0      0            0             0                 0  ...   
3        1      0            1             0                 1  ...   
4        0      0            0             0                 0  ...   

   other_infrastructure  weather_related  floods  storm  fire  earthquake  \
0                     0                0       0      0     0           0   
1                     0                1       0      1     0           0   
2                     0                0       0      0     0           0   
3                     0                0       0      0     0           0   
4                     0                0       0      0     0           0   

   cold  other_weather  direct_report  _merge  
0     0              0              0    both  
1     0              0              0    both  
2     0              0              0    both  
3     0              0              0    both  
4     0              0              0    both  

[5 rows x 41 columns]

In [19]:
# Are all messages in both tables?

df_raw['_merge'].value_counts()

both          26386
left_only         0
right_only        0
Name: _merge, dtype: int64

In [20]:
df_raw.drop(columns = '_merge', inplace = True)

In [21]:
df_raw.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 26386 entries, 0 to 26385
Data columns (total 40 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   id                      26386 non-null  int64 
 1   message                 26386 non-null  object
 2   original                10246 non-null  object
 3   genre                   26386 non-null  object
 4   related                 26386 non-null  int32 
 5   request                 26386 non-null  int32 
 6   offer                   26386 non-null  int32 
 7   aid_related             26386 non-null  int32 
 8   medical_help            26386 non-null  int32 
 9   medical_products        26386 non-null  int32 
 10  search_and_rescue       26386 non-null  int32 
 11  security                26386 non-null  int32 
 12  military                26386 non-null  int32 
 13  child_alone             26386 non-null  int32 
 14  water                   26386 non-null  int32 
 15  fo

# Empty columns

In [22]:
# What columns are empty?
df_raw.isna().mean()[df_raw.isna().mean() != 0]

original    0.611688
dtype: float64

In [23]:
# What are empty values in the 'original' column?

df_raw[df_raw['original'].isna()][['message','original']].head(15)

message original
7483           NOTES: It mark as not enough information      NaN
9956  My thoughts and prayers go out to all the live...      NaN
9957  I m sorry for the poor people in Haiti tonight...      NaN
9958  RT selenagomez UNICEF has just announced an em...      NaN
9959  lilithia yes 5.2 magnitude earthquake hit mani...      NaN
9960  RT TheNewsBlotter RT caribnews On Call Interna...      NaN
9961  Most Eureka homeowners won&#39 t collect on qu...      NaN
9962  Haiti hit by largest earthquake in over 200 ye...      NaN
9963  RT pinkelephantpun Earthquake Relief Donate ht...      NaN
9964  Praying that today s earthquake in #Haiti mini...      NaN
9965  RT YURFAVHALFBREED Moment of silence for those...      NaN
9966  RT lucasrohan Tem alguma coisa do Haiti? T u00...      NaN
9967  RT SamaritansPurse Currently mobilizing a team...      NaN
9968  My prayers to everyone affected by the earthqu...      NaN
9969  talking to ma mom..she lives in Jamaica still ...      NaN

# Duplicates

In [24]:
df_raw.duplicated().sum()

170

In [25]:
df_raw[df_raw.duplicated(keep = False)].head(5)

id                                            message  \
162  202  ?? port au prince ?? and food. they need gover...   
163  202  ?? port au prince ?? and food. they need gover...   
164  202  ?? port au prince ?? and food. they need gover...   
165  202  ?? port au prince ?? and food. they need gover...   
657  804  elle est vraiment malade et a besoin d'aide. u...   

                                              original   genre  related  \
162  p bay pap la syen ak manje. Yo bezwen ed gouve...  direct        1   
163  p bay pap la syen ak manje. Yo bezwen ed gouve...  direct        1   
164  p bay pap la syen ak manje. Yo bezwen ed gouve...  direct        1   
165  p bay pap la syen ak manje. Yo bezwen ed gouve...  direct        1   
657  she is really sick she need your help. please ...  direct        2   

     request  offer  aid_related  medical_help  medical_products  ...  \
162        1      0            1             0                 0  ...   
163        1      0            1             0                 0  ...   
164        1      0            1             0                 0  ...   
165        1      0            1             0                 0  ...   
657        0      0            0             0                 0  ...   

     aid_centers  other_infrastructure  weather_related  floods  storm  fire  \
162            0                     0                0       0      0     0   
163            0                     0                0       0      0     0   
164            0                     0                0       0      0     0   
165            0                     0                0       0      0     0   
657            0                     0                0       0      0     0   

     earthquake  cold  other_weather  direct_report  
162           0     0              0              0  
163           0     0              0              0  
164           0     0              0              0  
165           0     0              0              0  
657           0     0              0              0  

[5 rows x 40 columns]

In [26]:
df_raw.drop_duplicates(inplace = True)

# Investigating unique values in categories

In [27]:
for col in df_raw.select_dtypes(include = ['int32']):

    unique_values = df_raw[col].unique()

    # Check for columns that are not binary
    if len(unique_values) != 2:
        
        print(col)
        print(unique_values, end = '\n\n')

related
[1 0 2]

child_alone
[0]



It can be that `child_alone` might only not have a message labeled to that category.  
This is a labelling issue and will not be considered.

`related` on the other hand, should not have '2' in it. It will be further investigated.

In [28]:
df_raw[df_raw['related'] == 2].head()

id                                            message  \
117  146  Dans la zone de Saint Etienne la route de Jacm...   
221  263  . .. i with limited means. Certain patients co...   
307  373  The internet caf Net@le that's by the Dal road...   
462  565  Bonsoir, on est a bon repos aprs la compagnie ...   
578  700  URGENT CRECHE ORPHANAGE KAY TOUT TIMOUN CROIX ...   

                                              original   genre  related  \
117  Nan zon st. etine rout jakmel la bloke se mize...  direct        2   
221  t avec des moyens limites. Certains patients v...  direct        2   
307  Cyber cafe net@le ki chita rout de dal tou pr ...  direct        2   
462  Bonswa nou nan bon repo apri teleko nan wout t...  direct        2   
578  r et Salon Furterer. mwen se yon Cosmtologue. ...  direct        2   

     request  offer  aid_related  medical_help  medical_products  ...  \
117        0      0            0             0                 0  ...   
221        0      0            0             0                 0  ...   
307        0      0            0             0                 0  ...   
462        0      0            0             0                 0  ...   
578        0      0            0             0                 0  ...   

     aid_centers  other_infrastructure  weather_related  floods  storm  fire  \
117            0                     0                0       0      0     0   
221            0                     0                0       0      0     0   
307            0                     0                0       0      0     0   
462            0                     0                0       0      0     0   
578            0                     0                0       0      0     0   

     earthquake  cold  other_weather  direct_report  
117           0     0              0              0  
221           0     0              0              0  
307           0     0              0              0  
462           0     0              0              0  
578           0     0              0              0  

[5 rows x 40 columns]

They look like normal messages. Will be replaced with 1

In [29]:
df_raw['related'].replace(2,1, inplace = True)

# Investigating genres

In [30]:
df_raw['genre'].unique()

array(['direct', 'social', 'news'], dtype=object)

In [31]:
df_raw['genre'].value_counts(dropna = False)

news      13054
direct    10766
social     2396
Name: genre, dtype: int64

Genre looks OK. All categories are populated.

# Messages that are not alphabetical
Maybe some messages are completely unrelated to texts

In [32]:
df_raw[df_raw['message'].str.match(r'^\W+$')]

id                      message  \
7609    8497                 //// // @:@    
12293  14315                                
12297  14329                            .   
12330  14428                                

                                            original   genre  related  \
7609                               //// // @lt : gt@  direct        1   
12293                                              a  direct        1   
12297                                              a  direct        1   
12330           Shikarpur     a shikarpur          a  direct        1   

       request  offer  aid_related  medical_help  medical_products  ...  \
7609         0      0            0             0                 0  ...   
12293        0      0            0             0                 0  ...   
12297        0      0            0             0                 0  ...   
12330        0      0            0             0                 0  ...   

       aid_centers  other_infrastructure  weather_related  floods  storm  \
7609             0                     0                0       0      0   
12293            0                     0                0       0      0   
12297            0                     0                0       0      0   
12330            0                     0                0       0      0   

       fire  earthquake  cold  other_weather  direct_report  
7609      0           0     0              0              0  
12293     0           0     0              0              0  
12297     0           0     0              0              0  
12330     0           0     0              0              0  

[4 rows x 40 columns]

These look like wrong messages. Will be dropped

In [33]:
df_raw[df_raw['message'].str.match(r'^\W+$')].index

Int64Index([7609, 12293, 12297, 12330], dtype='int64')

In [34]:
df_raw.drop(df_raw[df_raw['message'].str.match(r'^\W+$')].index, inplace = True)

# Identifying not English messages

Different languages on the dataset can throw a model off.  
They will generate noise.  

Messages that don't have originals seem to be in english. This can be further investigated.  
There are messages that aren't in english on the message field. Id 804, 11200 for example

In [35]:
from langdetect import detect, DetectorFactory

In [36]:
DetectorFactory.seed = 123

In [37]:
def infer_lang(txt):

    try:

        lang = detect(txt)

    except:

        lang = None

    return lang

In [38]:
df_raw['infered_language'] = df_raw['message'].apply(infer_lang)

In [39]:
df_raw[df_raw['infered_language'].isna()]['message'].values

array(['(http://www.guardian.co.uk/global-development/2013/jan/16/somali-ngos-mogadishu-street-children)'],
      dtype=object)

Not being able to infer the language of a message that was actually a link.

In [40]:
df_raw[df_raw['infered_language'] != 'en'].shape

(406, 41)

In [41]:
df_raw[df_raw['infered_language'] != 'en'].sample(15, random_state= 1)

id                                            message  \
12425  14643  gaav valay aahar aor aarthik samashyaa saamna ...   
3289    3775         NOTES: personal request. Not an emergency.   
12284  14294  AURANG ZEB S/O GULAB DIN DIST MIANWALI TEH ESS...   
4697    5312           Please, help Lagonave. Help, Help,Help.    
6762    7619  I am a pupil at Mombin Crochu, Nord-est depart...   
12248  14186  KHALIL AHMAD ARIAN KOTLA MUGHLAN TAHSEL JAMPUR...   
15820  18522  (http://www.guardian.co.uk/global-development/...   
11853  13591  faltou energia aqui mais ja voltou. to indo pr...   
1256    1483  GEN YON TIBEBE KI MALAD NAN KOU PASKE BLOK TON...   
8129    9060       f k yo ap mal viv nan ansagal lagonav.msi.     
20684  23916                     Press inquiries: info@ofid.org   
14583  17182               Humanitarian Response to landslides:   
8766    9746  3 mamit chabon,4 dola poban,kart,pn 2 t,kreson...   
4165    4742                    WHEN SCHOOL GOING TE BE REOPEN    
8941    9981                   Zb rmd : dl 2, dl rep, dl chase    

                                                original   genre  related  \
12425  villagers r facing food and financial problm k...  direct        1   
3289                   Mwn vle yon ti infomasyon plis tj  direct        1   
12284     /            /            a,      a,        a.  direct        1   
4697                      MEZANMI WE SA NAFE POU LAGONAV  direct        1   
6762     MWEN MENM YON ELEV MOMBIN CROCHI DEPATMAN NODES  direct        1   
12248  KHALIL AHMAD ARIAN KOTLA MUGHLAN. TAHSIL JAMPU...  direct        1   
15820                                                NaN    news        1   
11853                                                NaN  social        1   
1256   VIRE NAN BO KKOTI YZIN NAN. WAP JWENN YON KAFO...  direct        1   
8129      f ke yo ap mal viv nan ansagale lagonav. mesi.  direct        1   
20684                                                NaN    news        0   
14583                                                NaN    news        1   
8766   3 mamit chabon, 4 dola poban, kart, pon 2 t, k...  direct        1   
4165   Bonswa, eske ouviti lekol la konfime pa minis ...  direct        1   
8941                     Zb rmd : dl 2, dl rep, dl chase  direct        1   

       request  offer  aid_related  medical_help  medical_products  ...  \
12425        0      0            0             0                 0  ...   
3289         0      0            0             0                 0  ...   
12284        0      0            0             0                 0  ...   
4697         1      0            1             0                 0  ...   
6762         0      0            0             0                 0  ...   
12248        0      0            0             0                 0  ...   
15820        0      0            0             0                 0  ...   
11853        0      0            0             0                 0  ...   
1256         0      0            0             0                 0  ...   
8129         0      0            0             0                 0  ...   
20684        0      0            0             0                 0  ...   
14583        0      0            1             0                 0  ...   
8766         0      0            0             0                 0  ...   
4165         0      0            0             0                 0  ...   
8941         0      0            0             0                 0  ...   

       other_infrastructure  weather_related  floods  storm  fire  earthquake  \
12425                     0                0       0      0     0           0   
3289                      0                0       0      0     0           0   
12284                     0                0       0      0     0           0   
4697                      0                0       0      0     0           0   
6762                      0                0       0      0     0           0   
12248            

The inferece looks like good enough. There are naturally mistakes, but dropping these rows might benefit more than do harm.

# Text cleaning

## Identifying links

In [42]:
df_raw[df_raw['message'].str.contains(r'http(s)?:\/\/[\w.-_]+\s', regex = True)]

C:\Users\Lucca\AppData\Local\Temp\ipykernel_17104\1027339002.py:1: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  df_raw[df_raw['message'].str.contains(r'http(s)?:\/\/[\w.-_]+\s', regex = True)]


id                                            message  \
5061    5737  If you want to find a Job at an NGO or the Gov...   
7403    8283            http://wap.sina.comhttp://wap.sina.com    
8913    9947  Nokia.com http://ea.mobile.nokia.com/ea/graphics    
9788   10960  BEGIN:VBKM VERSION:1.0 TITLE:Digicel Live Ha U...   
11470  13114  RT stro_Soichi: http://twitpic.com/15wu5u - Sa...   
...      ...                                                ...   
19592  22704  Dengue, an acute febrile disease spread by the...   
21924  25267  The UN Capital Development Fund (UNCDF) [ http...   
22037  25390  In a November overview, the UN Children's Fund...   
22659  26072  The map [ http://pakrelief.crowdmap.com ] on t...   
23879  27454  Over the years, there has been neglect of kala...   

                                                original   genre  related  \
5061   4636 : Si-w vle jwenn job nan yon ONG osinon g...  direct        0   
7403      http : //wap. sina. comhttp : //wap. sina. com  direct        1   
8913    Nokia.com http://ea.mobile.nokia.com/ea/graphics  direct        0   
9788   BEGIN:VBKM VERSION:1.0 TITLE:Digicel Live Ha U...  direct        1   
11470                                                NaN  social        1   
...                                                  ...     ...      ...   
19592                                                NaN    news        1   
21924                                                NaN    news        0   
22037                                                NaN    news        1   
22659                                                NaN    news        1   
23879                                                NaN    news        0   

       request  offer  aid_related  medical_help  medical_products  ...  \
5061         0      0            0             0                 0  ...   
7403         0      0            0             0                 0  ...   
8913         0      0            0             0                 0  ...   
9788         0      0            0             0                 0  ...   
11470        0      0            0             0                 0  ...   
...        ...    ...          ...           ...               ...  ...   
19592        0      0            1             1                 0  ...   
21924        0      0            0             0                 0  ...   
22037        0      0            0             0                 0  ...   
22659        1      0            1             0                 0  ...   
23879        0      0            0             0                 0  ...   

       other_infrastructure  weather_related  floods  storm  fire  earthquake  \
5061                      0                0       0      0     0           0   
7403                      0                0       0      0     0           0   
8913                      0                0       0      0     0           0   
9788                      0                0       0      0     0           0   
11470                     1                1       0      0     0           1   
...                     ...              ...     ...    ...   ...         ...   
19592                     0                0       0      0     0           0   
21924                     0                0       0      0     0           0   
22037                     0                1       1      0     0           0   
22659                     0                0       0      0     0           0   
23879                     0                0       0      0     0           0   

       cold  other_weather  direct_report  infered_language  
5061      0              0              0                en  
7403      0              0              0                en  
8913      0              0              0                pt  
9788      0              0              0                en  
11470     0              0              0                en  
...     ...            ...        

In [43]:
df_raw['message_cleaned'] = df_raw['message'].str.replace(r'http(s)?:\/\/[\w.-_]+\s', 'url_placeholder ', regex = True)

## Removing non alphabetical characters

In [44]:
pat_not_alnum = r'[^\w\s]'

In [45]:
df_raw[df_raw['message'].str.contains(pat_not_alnum, regex = True)]['message'].head()

0    Weather update - a cold front from Cuba that c...
3    UN reports Leogane 80-90 destroyed. Only Hospi...
4    says: west side of Haiti, rest of the country ...
5               Information about the National Palace-
7    Please, we need tents and water. We are in Sil...
Name: message, dtype: object

In [46]:
df_raw['message_cleaned'] = df_raw['message_cleaned'].str.replace(pat_not_alnum,' ',regex = True)

## Normalizing case

In [47]:
df_raw['message_cleaned'] = df_raw['message_cleaned'].str.lower()

In [48]:
df_raw.head()

id                                            message  \
0   2  Weather update - a cold front from Cuba that c...   
1   7            Is the Hurricane over or is it not over   
2   8                    Looking for someone but no name   
3   9  UN reports Leogane 80-90 destroyed. Only Hospi...   
4  12  says: west side of Haiti, rest of the country ...   

                                            original   genre  related  \
0  Un front froid se retrouve sur Cuba ce matin. ...  direct        1   
1                 Cyclone nan fini osinon li pa fini  direct        1   
2  Patnm, di Maryani relem pou li banm nouvel li ...  direct        1   
3  UN reports Leogane 80-90 destroyed. Only Hospi...  direct        1   
4  facade ouest d Haiti et le reste du pays aujou...  direct        1   

   request  offer  aid_related  medical_help  medical_products  ...  \
0        0      0            0             0                 0  ...   
1        0      0            1             0                 0  ...   
2        0      0            0             0                 0  ...   
3        1      0            1             0                 1  ...   
4        0      0            0             0                 0  ...   

   weather_related  floods  storm  fire  earthquake  cold  other_weather  \
0                0       0      0     0           0     0              0   
1                1       0      1     0           0     0              0   
2                0       0      0     0           0     0              0   
3                0       0      0     0           0     0              0   
4                0       0      0     0           0     0              0   

   direct_report  infered_language  \
0              0                en   
1              0                en   
2              0                no   
3              0                en   
4              0                en   

                                     message_cleaned  
0  weather update   a cold front from cuba that c...  
1            is the hurricane over or is it not over  
2                    looking for someone but no name  
3  un reports leogane 80 90 destroyed  only hospi...  
4  says  west side of haiti  rest of the country ...  

[5 rows x 42 columns]

## Removing double spacing

In [49]:
remove_double_spacing = lambda x: ' '.join(x.split())

In [51]:
df_raw['message_cleaned'] = df_raw['message_cleaned'].apply(remove_double_spacing)

df_raw['message_cleaned'] = df_raw['message_cleaned'].str.strip()

In [52]:
df_raw.head()

id                                            message  \
0   2  Weather update - a cold front from Cuba that c...   
1   7            Is the Hurricane over or is it not over   
2   8                    Looking for someone but no name   
3   9  UN reports Leogane 80-90 destroyed. Only Hospi...   
4  12  says: west side of Haiti, rest of the country ...   

                                            original   genre  related  \
0  Un front froid se retrouve sur Cuba ce matin. ...  direct        1   
1                 Cyclone nan fini osinon li pa fini  direct        1   
2  Patnm, di Maryani relem pou li banm nouvel li ...  direct        1   
3  UN reports Leogane 80-90 destroyed. Only Hospi...  direct        1   
4  facade ouest d Haiti et le reste du pays aujou...  direct        1   

   request  offer  aid_related  medical_help  medical_products  ...  \
0        0      0            0             0                 0  ...   
1        0      0            1             0                 0  ...   
2        0      0            0             0                 0  ...   
3        1      0            1             0                 1  ...   
4        0      0            0             0                 0  ...   

   weather_related  floods  storm  fire  earthquake  cold  other_weather  \
0                0       0      0     0           0     0              0   
1                1       0      1     0           0     0              0   
2                0       0      0     0           0     0              0   
3                0       0      0     0           0     0              0   
4                0       0      0     0           0     0              0   

   direct_report  infered_language  \
0              0                en   
1              0                en   
2              0                no   
3              0                en   
4              0                en   

                                     message_cleaned  
0  weather update a cold front from cuba that cou...  
1            is the hurricane over or is it not over  
2                    looking for someone but no name  
3  un reports leogane 80 90 destroyed only hospit...  
4  says west side of haiti rest of the country to...  

[5 rows x 42 columns]

# Export to SQL

In [69]:
conn = sqlalchemy.create_engine('sqlite:///' + os.path.abspath('../data/DisasterResponse.db'))

In [70]:
df_raw.to_sql('ANALYTICAL_TABLE', conn)

26212